src/config/ci.toml

```diff
@@ -17,7 +17,7 @@ stream_exchange_concurrent_barriers = 10
 imm_merge_threshold = 2
 
 [storage.object_store.retry]
-streaming_upload_attempt_timeout_ms = 10000
```

Comment:
This change will affect other tests, not just backfill test.

Reply:
Yes, but if this change is introduced to other tests, what will be the impact? I think they won’t break even if timeout is increased, even for aws sdk. Main reason I suggest this is to reduce maintenance workload

Reply:
The impact will be if there is any performance regression in the sdk, the chance we can catch it in CI will be lowered. For example, if for some reason the sdk is 10s slower in a new version in some corner cases, the CI test will still pass, while before this change, the CI test will fail.

There are four profiles used in `run-backfill-tests.sh`:
1. ci-3cn-1fe
2. ci-3cn-1fe-with-minio-rate-limit
3. ci-3cn-1fe-with-monitoring
4. ci-3cn-1fe-with-monitoring-and-minio-rate-limit

1 is used by many tests and 2-4 are used by backfill only. I think we can:
- Add one more toml files copied from `ci.toml` and change the timeout only.
- Change the 2-4 prefix from `ci` to `ci-backfill` and add one more risedev profile `ci-backfill-3cn-1fe`.
- Let the backfill test use the new profiles with the new toml config.